lib/katello/tasks/upgrades/3.9/migrate_sync_plans.rake

```diff
@@ -0,0 +1,24 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '3.9' do
+      desc "Migrate Pulp Sync Plans to new recurring logics"
+      task :migrate_sync_plans => ["environment"] do
+        User.current = User.anonymous_admin
+        puts "Starting recurring logic for migrated sync plans and deleting Pulp schedules"
+
+        Katello::SyncPlan.find_each do |sync_plan|
+          sync_plan.start_recurring_logic
+          sync_plan.foreman_tasks_recurring_logic.enabled = false unless sync_plan.enabled
+
+          sync_plan.products.each do |product|
+            product.repos(product.library).each do |repo_k|
+              repo = ::Katello::Repository.find(repo_k.id)
+              repo.sync_schedule(nil)
```

Comment:
We can't rely on this method to exist (In fact you should probably delete it).   Instead i would call into runcible directly with something like:

Katello.pulp_server.extensions.repository.remove_schedules(repo.pulp_id)